dmapiclient/data.py

```diff
@@ -183,6 +183,11 @@ def set_supplier_declaration(self, supplier_id, framework_slug, declaration, use
             user=user
         )
 
+    def remove_supplier_declaration(self, supplier_id, framework_slug, user):
+        return self.set_supplier_declaration(
+            supplier_id, framework_slug, declaration={}, user=user
+        )
```

Comment:
This calls a different view - PUT instead of POST. We should be able to do something like

`return self._post_with_updated_by(<the same url>, data={}, user=user)`